subcommands/devices/show.go

```diff
@@ -49,6 +49,11 @@ func doShow(cmd *cobra.Command, args []string) {
 	fmt.Printf("Up to date:\t%v\n", device.UpToDate)
 	fmt.Printf("Target:\t\t%s / sha256(%s)\n", device.TargetName, device.OstreeHash)
 	fmt.Printf("Ostree Hash:\t%s\n", device.OstreeHash)
+	if target, errTarget := api.TargetGet(factory, device.TargetName); errTarget == nil {
+		if custom, errCustom := api.TargetCustom(*target); errCustom == nil {
+			fmt.Printf("LmP Version:\t%s\n", custom.LmpVer)
+		}
+	}
```

Comment:
We do fetch the targets for the device already in the API call to get device info.
So, I'd prefer to extend the API to return this additional attribute, rather than calling for targets again.

**TL;DR**
There are more problems to this API call than you might think.
For example, if this is a production device which runs e.g. target `foo-42`, and that target was already pruned from the CI targets due to its age - an API can still return its lmp-ver from prod targets.  You'd need to fully duplicate parts of the current API logic in Fioctl to achieve the same.

Reply:
>We do fetch the targets for the device already in the API call to get device info.

I actually was looking for it here https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/api/device.py#L916 and couldn't find. Please, point me to the line where it happens.

Reply:
>For example, if this is a production device which runs e.g. target foo-42, and that target was already pruned from the CI targets due to its age - an API can still return its lmp-ver from prod targets.

I guess, I can add another call to the prod targets if 404 is returned to the initial/CI call.

Reply:
[Here](https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/api/device.py#L236) and [here](https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/api/device.py#L259).

You've got the idea.

Reply:
It seems easier just leave it as it is now :).

Trying to adjust `FactoryVersions` for this use-case seems like too much. I am not sure I understand how I can get the `custom.lmp-ver` field if I have just a target name as an input (the custom fields are not stored).